xmtp_mls/src/subscriptions/stream_all.rs

```diff
@@ -105,6 +119,17 @@ where
         let mut this = self.as_mut().project();
 
         if let Ready(msg) = this.messages.as_mut().poll_next(cx) {
+            if let Some(Ok(msg)) = &msg {
+                if self.sync_groups.contains(&msg.group_id) {
```

Comment:
Couple questions:

instead of piping to an intermediary channel here, why not just call stream all messages with the filter set to only sync groups?

Also, if there are multiple streams running, how do we ensure the sync worker isn't just getting a bunch of duplicate messages


This stream will also return the sync group message since its `return Ready(msg)`, could that be possibly confusing if someone streams all messages, and now they're getting sync messages in two places.

Reply:
That's a good concern. I'm a little out of context with streaming, but thinking about this a bit more, the integrator should never receive sync group messages. I've adjusted the handling to filter out sync messages and just notify the worker [here](https://github.com/xmtp/libxmtp/pull/1961/commits/8f85dc469645cf42aad552a5c85fa815b4815269#diff-4a3d32ed6005867d963b0eae023dc9c1fd8ffd9531413ba572f4287210a18cc1R131).

If the worker receives multiple pings from multiple streams, that's fine. The worker will receive the ping, see that there are no new unprocessed messages and go back to sleep.